src/run_graph.rs

```diff
@@ -105,14 +113,17 @@ impl TasksGraph {
         if dependencies == 0 {
             match self.graph[node] {
                 Node::Task(_) => {
-                    let content = mem::replace(&mut self.graph[node], Node::RunningTask);
+                    let content = mem::replace(&mut self.graph[node], Node::RunningTask(None));
                     if let Node::Task(task) = content {
+                        let task = Arc::new(task);
```

Comment:
Generally speaking this looks like an odd thing to me -- should we be constructing the Arc somewhere more at the initial place tasks get created?

Reply:
Sure, done.